auth/token.go

```diff
@@ -23,10 +23,14 @@ import (
 )
 
 const (
-	// expirationDelta is a margin of error during which a token should be
-	// considered expired. This should help avoid expiration races when server
-	// times don't match
+	// expirationDelta and ClockDriftDelta are margins of error during
+	// which a token should be considered expired.
+	// Together they help avoid expiration races when server/host times don't match
 	ExpirationDelta = 10 * time.Second
+	ClockDriftDelta = 5 * time.Minute
+
+	// refresh
+	RefreshAhead = 5 * time.Minute
```

Comment:
Can we get separate comments on each of these explaining exactly what they control:  ClockDrift gives leeway on request expiration, ExpirationDelta gives leeway on token expiration, and RefreshAhead tells the delegate to refresh its auth token well in advance of expiration.